### PR TITLE
GVT-2154 Review fixes, TODO-cleanup, add test for using duplicate track as a base for target

### PR DESCRIPTION
**Pull request by jyrkija**

### CODE REVIEW

**Review by solita-mrunsten — COMMENTED** [review]
👍

File: infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt [hunk 1]
```diff
@@ -1212,13 +1193,17 @@ class SwitchLinkingService @Autowired constructor(
         }
     }
 
+    @Transactional(readOnly = true)
+    fun getTrackSwitchSuggestions(publishType: PublishType, trackId: IntId<LocationTrack>) =
+        getTrackSwitchSuggestions(locationTrackDao.getOrThrow(publishType, trackId))
+
     @Transactional(readOnly = true)
     fun getTrackSwitchSuggestions(track: LocationTrack): List<Pair<IntId<TrackLayoutSwitch>, SuggestedSwitch?>> {
```
Comment (juuseko) [inline infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt:1201]: Puuttuukos tältä muuten tarkoituksella lokitus?

File: infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt [hunk 2]
```diff
@@ -325,10 +326,11 @@ private fun createSplitTarget(
...
         descriptionSuffix = request.descriptionSuffix,
 
+        externalId = null, // New track -> no external ID
+        // After split, tracks are not duplicates
         duplicateOf = null,
-        externalId = null,
-        topologyStartSwitch = null,
-        topologyEndSwitch = null,
+        topologyStartSwitch = null, // Topology is re-resolved after tracks and switches are updated
+        topologyEndSwitch = null, // Topology is re-resolved after tracks and switches are updated
```
Comment (juuseko) [inline infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt:333]: Todella nitpick, mutta itse pistäisin mielummin yhden kommentin näiden molempien rivien yläpuolelle kuin kaksi erillistä

**Review by juuseko — COMMENTED** [review]
👍

File: infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt (hunk 1 above)
Comment (jyrkija) [reply infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt:1201]: Ei, vahingoltahan tuo näyttää. Lisään.

File: infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt (hunk 2 above)
Comment (jyrkija) [reply infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt:333]: Joo toisessa kohtaa se onkin niin. Vaihdan tuonkin.
